opm/parser/eclipse/Deck/Deck.hpp

```diff
@@ -113,17 +115,24 @@ namespace Opm {
             const UnitSystem& getDefaultUnitSystem() const;
             const UnitSystem& getActiveUnitSystem() const;
 
+            const std::string getDataFile() const;
+            void setDataFile(const std::string& dataFile);
+            bool hasDataFile() const;
 
         private:
             Deck( std::vector< DeckKeyword >&& );
 
             void initUnitSystem() const;
 
+
             std::vector< DeckKeyword > keywordList;
             mutable MessageContainer m_messageContainer;
 
             mutable std::unique_ptr< UnitSystem > activeUnits;
             mutable std::unique_ptr< UnitSystem > defaultUnits;
+
+            std::string m_dataFile;
+            bool m_hasDataFile;
```

Comment:
Good input, simplified with new commit.
1. No data means deck is created from string, not file
2. Now only sets filename in deck after successfully opened by Parser. No check inside deck for validity, should we there as well?
3. This should not happen. see 2.